### DISCUSSION TITLE
wis2box 1.0b5 released

### DISCUSSION THREAD

**Discussion by tomkralidis** [wmo-im/wis2box#542] [Announcements]
This is the fifth 1.0 beta release of wis2box. This release includes a number of new features, enhancements and bugfixes.

## Updates

wis2box-webapp (new container) provides an additional web application adding the following functionality:
- new station editor user interface with support to query OSCAR/Surface
- data entry via web-based form for SYNOP ASCII and CSV
- monitoring dashboard to via data notifications and download/inspect BUFR produced by wis2box

Updates to support wis2box configuration:
- new configuration creation script to allow the user to a create initial `wis2box.env`, `data-mappings.yml` and template discovery metadata files by answering a series of question on the command line
- environment files `default.env` and `dev.env` have been replaced by `wis2box.env`

Enhancements:
- add barometer height to station metadata management
- enable adding more user to default mosquitto broker
- fix to generate valid WNM (base64 integrity)
- fixes to generate valid WCMP2 (needs flushing of ES discovery-metadata)
- support for Docker Compose v2 (Note is it required or supported?)
- CSV publication (csv2bufr):
  - Improved logging and return of warnings and errors to the result object
  - BUFR templates now managed though the `csv2bufr.templates` module (note this will be moved to a stand alone module in a future release)
  - addition of minimal template for data from AWS weather stations
 
Bug fixes:
- fix dataset deletion workflow
- do not remove discovery metadata when running data cleaning
- BUFR publication: BUFR headers are now correctly set in extracted subsets. Previously the headers were not copied, leading to incorrect replication factors and data corruption
- GeoJSON API publication: code table entry description now correctly extracted and set in GeoJSON output. Previously only the first word of the description was extracted

  
## important

- use the station editor in wis2box-webapp and ensure your stations are associated to a topic to ensure you can see the data in the "MAP" and "EXPLORE" option of wis2box-ui
- the cronjob running `wis2box metadata station publish-collection` in the wis2box-management every 10 minutes is now disabled

## Migrating existing installations

The following steps are required when upgrading from wis2box **1.0b4** to **1.0b5**:

```bash
# stop current wis2box
cd ~/wis2box-1.0b4 && python3 wis2box-ctl.py stop
# prune unused docker images
docker image prune -a
# delete api-config volume
docker volume rm wis2box_project_api-config
# download new release in home-directory
cd && wget https://github.com/wmo-im/wis2box/releases/download/1.0b5/wis2box-setup-1.0b5.zip
# unzip and go to new directory
unzip wis2box-setup-1.0b5.zip && cd wis2box-1.0b5
# run migration script to create wis2box.env
python3 migration/migration_create_wis2box_env.py
# check content of wis2box.env correctly copied your local updates in dev:env
cat wis2box.env
# run migration script to add barometer_height column to station_list.csv:
python3 migration/migration_update_station_list.py
# start wis2box
python3 wis2box-ctl.py start
# login to wis2box and execute commands
python3 wis2box-ctl.py login
# (inside wis2box-management container)
wis2box data add-collection /data/wis2box/surface-weather-observations.yml
wis2box metadata station publish-collection
```

**For installations older than 1.0b4, it is recommended to start from a clean/fresh installation.**

To start a fresh installation, please clean your docker system after stopping your wis2box:

```bash
python3 wis2box-ctl.py stop
docker image prune -a
docker volume prune -a
```

...then delete the directory containing your current release and download the latest release and following the setup instructions.

## Documentation
Documentation for this release is available at https://docs.wis2box.wis.wmo.int/en/1.0b5

## Feedback
All contributions are always welcome. Feedback is welcome via GitHub [issues](https://github.com/wmo-im/wis2box/issues) or [discussions](https://github.com/wmo-im/wis2box/discussions).

The wis2box team

https://docs.wis2box.wis.wmo.int

**Comment by isedwards** [comment]
Could you update the migration instructions here, and in the release notes, to add `stop` after the `wis2box-ctl.py`?

```
# stop current wis2box
cd ~/wis2box-1.0b4 && python3 wis2box-ctl.py
```

**Reply by tomkralidis** [reply]
Good catch @isedwards! Fixed.

**Comment by eyou81** [comment]
hello , i have this problem
root@e870d1bd98d5:/home/wis2box# wis2box metadata stations publish-collections
Usage: wis2box metadata [OPTIONS] COMMAND [ARGS]...
Try 'wis2box metadata --help' for help.

Error: No such command 'stations'.
